### PR TITLE
feat: Create and Vote on Token Transfer Proposal

### PR DESCRIPTION
**Pull request by damiano-melcarne-lab49**
**Description**:

- Updated Token Transfer Proposal form fields to match new proposal contract.
- Added validations for Token Transfer Proposal form fields.
- Added addresses for new proposal contacts (see `GovernorProxyContracts`).
- Add functionality to create and vote on a Transfer Token proposals.

**Tech Notes**
- Added `react-hook-form` library to handle form validations.

**Screenshots**:
Screen Shot 2022-12-13 at 10 16 17 PM
Screen Shot 2022-12-13 at 10 16 40 PM

### CODE REVIEW

File: src/dex-ui/pages/CreateProposal/CreateProposal.tsx [hunk 1]
```diff
@@ -38,29 +24,12 @@ export const CreateProposal = (props: CreateProposalProps) => {
    * In case the user bookmarks the URL
    * */
   const { proposalType } = props;
-
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/CreateProposal/CreateProposal.tsx:41]: Form state is now managed within the Proposal Form components using `react-hook-form`.

File: src/dex-ui/pages/ProposalDetails/ProposalDetails.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ export const ProposalDetails = () => {
 
   const castVote = (voteType: VoteType) => {
     if (formattedProposal) {
-      governance.castVote(formattedProposal.id, voteType);
+      governance.castVote(formattedProposal.contractId, formattedProposal.id, voteType);
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/ProposalDetails/ProposalDetails.tsx:46]: We need the proposals contractId to cast the vote on the correct proposal contract.

File: src/dex-ui/services/HederaService/HederaService.ts [hunk 3]
```diff
@@ -202,23 +202,10 @@ function createHederaService() {
...
-      .setFunction(GovernorContractFunctions.CastVote, contractFunctionParameters)
-      .setGas(900000)
-      .setNodeAccountIds([new AccountId(3)])
-      .freezeWithSigner(signer);
-
-    const response = await castVoteTransaction.executeWithSigner(signer);
-    return response;
-  };
-
   const fetchProposalState = async (proposalId: BigNumber): Promise<BigNumber | undefined> => {
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/HederaService/HederaService.ts:205]: These functions will need to be updated to handle dynamic proposal contract ids.

File: src/dex-ui/store/walletSlice/walletSlice.ts [hunk 4]
```diff
@@ -23,12 +23,12 @@ const initialWalletState: WalletState = {
 const createWalletSlice: WalletSlice = (set, get): WalletStore => {
   return {
     ...initialWalletState,
-    getTokenAmountWithPrecision: (tokenSymbol: string, tokenAmount: number) => {
+    getTokenAmountWithPrecision: (tokenSymbol: string, tokenAmount: number, tokenId?: string) => {
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/store/walletSlice/walletSlice.ts:26]: This is a temporary fix to get the token amount to precision based on the tokenId.

**Review by web3-nomad — APPROVED** [review]

**Pull request merged**